zbus_xmlgen/src/lib.rs

```diff
@@ -294,3 +314,54 @@ pub fn pascal_case(s: &str) -> String {
     }
     pascal
 }
+
+fn estimate_type_complexity(it: &mut std::iter::Peekable<std::slice::Iter<'_, u8>>) -> u32 {
+    let mut score = 0;
+    let c = it.next().unwrap();
+    match *c as char {
+        u8::SIGNATURE_CHAR
+        | bool::SIGNATURE_CHAR
+        | i16::SIGNATURE_CHAR
+        | u16::SIGNATURE_CHAR
+        | i32::SIGNATURE_CHAR
+        | u32::SIGNATURE_CHAR
+        | i64::SIGNATURE_CHAR
+        | u64::SIGNATURE_CHAR
+        | f64::SIGNATURE_CHAR
+        | <&str>::SIGNATURE_CHAR => {
+            score += 1;
+        }
+        'h' => score += 10,
+        Signature::SIGNATURE_CHAR | VARIANT_SIGNATURE_CHAR | ObjectPath::SIGNATURE_CHAR => {
+            score *= 10
+        }
+        ARRAY_SIGNATURE_CHAR => {
+            let c = it.peek().unwrap();
+            match **c as char {
+                '{' => {
+                    score *= 10;
+                    score += estimate_type_complexity(it);
+                }
+                _ => {
+                    score += 5 * estimate_type_complexity(it);
+                }
+            }
+        }
+        STRUCT_SIG_START_CHAR | DICT_ENTRY_SIG_START_CHAR => {
+            score += 50;
+            loop {
+                let c = it.peek().unwrap();
+                match **c as char {
+                    STRUCT_SIG_END_CHAR | DICT_ENTRY_SIG_END_CHAR => {
+                        // consume the closing character
+                        it.next().unwrap();
+                        break;
+                    }
+                    _ => score += 5 * estimate_type_complexity(it),
+                }
+            }
+        }
+        _ => {}
+    };
+    score
+}
```

Comment:
This code is a bit complicated so it would be great to have tests for this. Just add the cases to the existing test.